### PR TITLE
HF audit actions reducers

### LINKED ISSUE #470: HF audit zkApp docs: o1js Actions & Reducer actions-and-reducer.mdx
Audit and update all zkApp docs for technical accuracy prior to hard fork #459
Before the HF, verify technical content in 

Actions & Reducer
https://docs.minaprotocol.com/zkapps/o1js/actions-and-reducer

123 lines
https://github.com/o1-labs/docs2/blob/main/docs/zkapps/o1js/actions-and-reducer.mdx

### PR DESCRIPTION
**Pull request by ymekuria**
Closes #470

### CODE REVIEW

File: docs/zkapps/writing-a-zkapp/feature-overview/actions-and-reducer.mdx [hunk 1]
```diff
@@ -100,25 +99,37 @@ A zkApp can retrieve events and actions from one or more Mina archive nodes. If
...
 ```ts
-reducer = Reducer({ actionType: AsFieldElements<A> });
-
-this.reducer.dispatch(action: A): void;
+reducer = Reducer({ actionType: FlexibleProvablePure<Action> });
+
+this.reducer.dispatch(action: Action): void;
+
+this.reducer.reduce<State>(
+  actions: MerkleList<MerkleList<Action>>,
```
Comment (MartinMinkov) [inline docs/zkapps/writing-a-zkapp/feature-overview/actions-and-reducer.mdx:107]: Just checking if the double nested `MerkleList is the right usage?

File: docs/zkapps/writing-a-zkapp/feature-overview/actions-and-reducer.mdx [hunk 2]
```diff
@@ -100,25 +99,37 @@ A zkApp can retrieve events and actions from one or more Mina archive nodes. If
...
-  stateType: AsFieldElements<S>,
-  reduce: (state: S, action: A) => S,
-  initial: { state: S, actionState: Field }
-): { state: S, actionState: Field };
 
-Reducer.initialActionState: Field;
 ```
 
-In the near future, we want to add a function to retrieve actions from an archive node:
+The `getActions` function retrieves a list of previoulsy emmited actions from an archive node:
```
Comment (MartinMinkov) [inline docs/zkapps/writing-a-zkapp/feature-overview/actions-and-reducer.mdx:121]: `previoulsy emmited` spelling

File: docs/zkapps/writing-a-zkapp/feature-overview/actions-and-reducer.mdx [hunk 3]
```diff
@@ -73,23 +73,22 @@ let stateType = Bool;
...
-  actionState: Reducer.initialActionState,
 };
 
-let { state, actionState } = this.reducer.reduce(
+let newState = this.reducer.reduce(
   actions,
   stateType,
   (state: Bool, action: Field) => state.or(action.equals(1000)),
-  initial
+  initial
```
Comment (MartinMinkov) [inline docs/zkapps/writing-a-zkapp/feature-overview/actions-and-reducer.mdx:85]: additional space added here

File: docs/zkapps/writing-a-zkapp/feature-overview/actions-and-reducer.mdx (hunk 1 above)
Comment (ymekuria) [reply docs/zkapps/writing-a-zkapp/feature-overview/actions-and-reducer.mdx:107]: Here is the reduce definition. 

https://github.com/o1-labs/o1js/blob/a8b9b5e761599a8168ec858b3671ffbd1e771fe0/src/lib/mina/actions/reducer.ts#L70

**Review by MartinMinkov — APPROVED** [review]

**Pull request merged**